scylla-server/src/services/data_service.rs

```diff
@@ -1,80 +1,70 @@
-use prisma_client_rust::QueryError;
-
-use crate::{prisma, processors::ClientData, Database};
-
-prisma::data::select! {public_data {
-    time
-    values
-}}
+use crate::{models::Data, schema::data::dsl::*, ClientData, Database};
+use diesel::prelude::*;
 
 /// Get datapoints that mach criteria
-/// * `db` - The prisma client to make the call to
+/// * `db` - The database connection to use
 /// * `data_type_name` - The data type name to filter the data by
 /// * `run_id` - The run id to filter the data
-/// * `fetch_run` whether to fetch the run assocaited with this data
-/// * `fetch_data_type` whether to fetch the data type associated with this data
-///   returns: A result containing the data or the QueryError propogated by the db
+///   returns: A result containing the data or the error propogated by the db
 pub async fn get_data(
-    db: &Database,
+    db: &mut Database,
     data_type_name: String,
     run_id: i32,
-) -> Result<Vec<public_data::Data>, QueryError> {
-    db.data()
-        .find_many(vec![
-            prisma::data::data_type_name::equals(data_type_name),
-            prisma::data::run_id::equals(run_id),
-        ])
-        .select(public_data::select())
-        .exec()
-        .await
+) -> Result<Vec<Data>, diesel::result::Error> {
+    data.filter(runId.eq(run_id).and(dataTypeName.eq(data_type_name)))
+        .load(db)
 }
 
 /// Adds a datapoint
-/// * `db` - The prisma client to make the call to
+/// * `db` - The database connection to use
 /// * `serverdata` - The protobuf message to parse, note the unit is ignored!
 /// * `unix_time` - The time im miliseconds since unix epoch of the message
 /// * `data_type_name` - The name of the data type, note this data type must already exist!
 /// * `rin_id` - The run id to assign the data point to, note this run must already exist!
 ///   returns: A result containing the data or the QueryError propogated by the db
 pub async fn add_data(
-    db: &Database,
+    db: &mut Database,
     client_data: ClientData,
-) -> Result<public_data::Data, QueryError> {
-    db.data()
-        .create(
-            prisma::data_type::name::equals(client_data.name),
-            client_data.timestamp.fixed_offset(),
-            prisma::run::id::equals(client_data.run_id),
-            vec![prisma::data::values::set(
-                client_data.values.iter().map(|f| *f as f64).collect(),
-            )],
-        )
-        .select(public_data::select())
-        .exec()
-        .await
+) -> Result<Data, diesel::result::Error> {
```

Comment:
Has one of y'all double checked that each of the endpoints specifically still work as expected (new run, add data, etc.), just wanna make sure

Reply:
tested minimally... but it at the very least that works as expected